WeatherApp/Controller/LocationWeatherViewController.swift

```diff
@@ -0,0 +1,172 @@
+//
+//  LocationWeatherViewController.swift
+//  WeatherApp
+//
+//  Created by Marcin Wójciak on 17/01/2020.
+//  Copyright © 2020 Marcin Wójciak. All rights reserved.
+//
+
+import UIKit
+import CoreLocation
+
+class LocationWeatherViewController: UITableViewController, CLLocationManagerDelegate, WeatherManagerDelegate {
+    
+    let userDefaults = UserDefaults.standard
+    let locationManager = CLLocationManager()
+    var weatherManager = WeatherManager()
+    
+    var userLocations = [String]()
+    var weatherData = [WeatherModel]()
+    
+    override func viewWillAppear(_ animated: Bool) {
+        
+        
+    }
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        tableView.register(UINib(nibName: "LocationWeatherCell", bundle: nil), forCellReuseIdentifier: "LocationWeatherCell")
+        
+        weatherManager.delegate = self
+        
+        locationManager.requestWhenInUseAuthorization()
+        locationManager.desiredAccuracy = kCLLocationAccuracyThreeKilometers
+        locationManager.delegate = self
+        locationManager.requestLocation()
+        
+        if userLocations.count == 0 {
+            userLocations = userDefaults.stringArray(forKey: "UserLocations") ?? []
+        }
+        print(userLocations)
+        
+        for location in userLocations {
+            weatherManager.fetchWeatherData(for: location)
+        }
+        
+        // Uncomment the following line to preserve selection between presentations
+        // self.clearsSelectionOnViewWillAppear = false
+        
+    }
+    
+    override func viewWillDisappear(_ animated: Bool) {
+        
+    }
+    
+    
+    //MARK: - Add User Location
+    
+    @IBAction func addLocationPressed(_ sender: UIBarButtonItem) {
+        
+        var locationName = UITextField()
+        
+        let alert = UIAlertController(title: "Add New User Location", message: "Please type city name", preferredStyle: .alert)
+        
+        alert.addTextField { (textField) in
+            textField.placeholder = "City Name"
+            locationName = textField
+        }
+        
+        let addUserLocation = UIAlertAction(title: "Add", style: .default) { (addAction) in
+            
+            guard let locationNameString = locationName.text?.trimmingCharacters(in: .whitespacesAndNewlines) else {
+                return
+            }
+            
+            guard !self.userLocations.contains(locationNameString) else {
+                return
+            }
+            
+            if locationNameString != "" {
```

Comment:
Do `String`a możemy też używać `.isEmpty` zamiast porównywania do pustego `String`a. Unikamy tworzenia nowego obiektu, bo `""` to de facto stworzenie nowego obiektu.